R/setops.R

```diff
@@ -11,16 +11,18 @@ validate <- function(cols, dt) {
   cols = as.integer(cols)
   isna = which(!cols %in% seq_along(dt))
   if (length(isna))
-    stop(argcols, " value(s) [", paste(origcols[isna], collapse=", "), "] not present (or out of range) in ", argdt)
+    stop(argcols, " value", if (length(isna) > 1L) 's ' else ' ',
+         brackify(origcols[isna]), " not present (or out of range) in ", argdt)
   cols
 }
 
 # setdiff for data.tables, internal at the moment #547, used in not-join
 setdiff_ <- function(x, y, by.x=seq_along(x), by.y=seq_along(y), use.names=FALSE) {
   if (!is.data.table(x) || !is.data.table(y)) stop("x and y must both be data.tables")
-  if (is.null(x) || !length(x)) return(x)
```

Comment:
I guess this was meant to capture `null.data.table()`? `NULL` itself is caught by `is.data.table`...